### ISSUE TITLE
It’s impossible to run Twisted test suite with in-dev pyOpenSSL

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8056]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#8056|
|Type|enhancement|
|Created|2015-10-20 09:04:53Z|
|Branch|https://github.com/twisted/twisted/tree/tolerate-pyopenssl-suffix-8056|

In-dev pyOpenSSLs can be recognized by a .dev0 suffix to their version.  Twisted can't handle that though which makes it impossible for pyOpenSSL to run the Twisted test suite as part of it's CI to verify we haven’t broken anything.  See also https://github.com/pyca/pyopenssl/pull/374

<details><summary>Searchable metadata</summary>

```
trac-id__8056 8056
type__enhancement enhancement
reporter__hynek hynek
priority__normal normal
milestone__None None
branch__branches_tolerate_pyopenssl_suffix_8056 branches/tolerate-pyopenssl-suffix-8056
branch_author__hynek hynek
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1445331893854097 1445331893854097
changetime__1445421660492972 1445421660492972
version__None None
owner__hynek hynek

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_1"></a>|@hynek commented|
|-|-|

(In [45867]) Branching to tolerate-pyopenssl-suffix-8056.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_2"></a>|@hynek commented|
|-|-|

buildbots are green, cheap review points! :)

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa commented|
|-|-|

Not doing a full review because I don't know enough Twisted to be useful, but I don't think you need to conversion to `list()`: Python can safely tuple-unpack a generator.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_4"></a>|@hynek commented|
|-|-|

still green :)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @hynek|
|-|-|

A masterpiece.

Please land.

Thanks for doing this - it has been really annoying when developing treq.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_6"></a>|@hynek set status to `closed`|
|-|-|

(In [45877]) Merge tolerate-pyopenssl-suffix-8056

Author: hynek
Reviewer: glyph
Fixes: [#8056](https://github.com/twisted/twisted/issues/8056)

Tolerate suffixes to pyOpenSSL's version string.
